app/models/concerns/stocktake_processor.rb

```diff
@@ -0,0 +1,76 @@
+module StocktakeProcessor
+  extend ActiveSupport::Concern
+
+  included do
+    private_class_method :apply_package_revision
+  end
+
+  class_methods do
+    #
+    # Iterates through the stocktake revisions and applies changes
+    #
+    # @param [Stocktake] stocktake the stocktake instance to process
+    #
+    # @return [boolean]
+    #
+    def process_stocktake(stocktake)
+      errors = []
+
+      raise Goodcity::InvalidStateError.new(I18n.t('stocktakes.invalid_state')) unless stocktake.open?
+      raise Goodcity::InvalidStateError.new(I18n.t('stocktakes.dirty_revisions')) if stocktake.revisions.where(dirty: true).count.positive?
+
+      PackagesInventory.secured_transaction do
```

Comment:
can be moved to model or a different method, just to avoid a big method

Reply:
most of the processing has already been moved to `apply_package_revision`. This is just a loop that calls that method, the function looks big because I add a lot of spacing between lines